webapp/src/Controller/Team/SubmissionController.php

```diff
@@ -135,14 +139,15 @@ public function createAction(Request $request)
      * @Route("/submission/{submitId<\d+>}", name="team_submission")
      * @param Request $request
      * @param int     $submitId
-     * @return \Symfony\Component\HttpFoundation\Response
-     * @throws \Doctrine\ORM\NonUniqueResultException
+     * @return Response
+     * @throws NonUniqueResultException
      */
     public function viewAction(Request $request, int $submitId)
     {
         $verificationRequired = (bool)$this->config->get('verification_required');;
         $showCompile      = $this->config->get('show_compile');
         $showSampleOutput = $this->config->get('show_sample_output');
+        $allowDownload    = (bool)$this->config->get('allow_team_submission_download');
```

Comment:
what happens if that field does not exist?

Reply:
Then it returns false since that is in the yaml

Reply:
The yaml is only imported if people do a db upgrade, IIRC? If they only update the code it doesn't exist in the DB, right?

Reply:
You are not right 🤣. We always read the yaml from the yaml. The db only has values if they differ from the yaml (I.e. if you change them in the UI)

Reply:
Even better, where do we read the yaml?

Reply:
In the configuration service.  ie in the function that gets called here